staging/src/k8s.io/component-helpers/scheduling/corev1/helpers.go

```diff
@@ -30,3 +34,102 @@ func PodPriority(pod *v1.Pod) int32 {
 	// name of the pod was empty. So, we resolve to the static default priority.
 	return 0
 }
+
+// NodeSelectorRequirementsAsSelector converts the []NodeSelectorRequirement api type into a struct that implements
+// labels.Selector.
+func NodeSelectorRequirementsAsSelector(nsm []v1.NodeSelectorRequirement) (labels.Selector, error) {
+	if len(nsm) == 0 {
+		return labels.Nothing(), nil
+	}
+	selector := labels.NewSelector()
+	for _, expr := range nsm {
+		var op selection.Operator
+		switch expr.Operator {
+		case v1.NodeSelectorOpIn:
+			op = selection.In
+		case v1.NodeSelectorOpNotIn:
+			op = selection.NotIn
+		case v1.NodeSelectorOpExists:
+			op = selection.Exists
+		case v1.NodeSelectorOpDoesNotExist:
+			op = selection.DoesNotExist
+		case v1.NodeSelectorOpGt:
+			op = selection.GreaterThan
+		case v1.NodeSelectorOpLt:
+			op = selection.LessThan
+		default:
+			return nil, fmt.Errorf("%q is not a valid node selector operator", expr.Operator)
+		}
+		r, err := labels.NewRequirement(expr.Key, op, expr.Values)
+		if err != nil {
+			return nil, err
+		}
+		selector = selector.Add(*r)
+	}
+	return selector, nil
+}
+
+// MatchNodeSelectorTerms checks whether the node labels and fields match node selector terms in ORed;
+// nil or empty term matches no objects.
+func MatchNodeSelectorTerms(
```

Comment:
can we change this to `MatchNodeSelectorTerms(node *v1.Node, terms []v1.NodeSelectorTerm) bool`?

Reply:
```
pkg/scheduler/framework/plugins/helper/node_affinity.go:75:	return corev1.MatchNodeSelectorTerms(nodeSelectorTerms, node.Labels, fields.Set{
```
relies on `nodeFields fields.Set` argument to be set.

Reply:
Really though, the context of that usage is:

```go
// nodeMatchesNodeSelectorTerms checks if a node's labels satisfy a list of node selector terms,
// terms are ORed, and an empty list of terms will match nothing.
func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
	return corev1.MatchNodeSelectorTerms(nodeSelectorTerms, node.Labels, fields.Set{
		"metadata.name": node.Name,
	})
}
```

So it's just a wrapper using the same signature @alculquicondor is suggesting.  The only other usage of this (in `pkg/volume/util` sets it to nil:
```
pkg/volume/util/util.go:180:		if !corev1.MatchNodeSelectorTerms(terms, labels.Set(nodeLabels), nil) {
```

The question is do we want to switch the intent of this function to check *all* of the node's fields/labels or leave the option to pass specific ones to check against?

Reply:
isn't it rather none when the fields are `nil`?
In any case, this is a question for pkg/volume OWNERs

Reply:
> isn't it rather none when the fields are nil?

It checks against whatever is passed in. If that's nil, then it's none, but it could be a specific subset

Reply:
Where you able to contact sig-storage regarding this?

Reply:
> Where you able to contact sig-storage regarding this?

I'm not sure who to ping on that (@kubernetes/sig-storage-misc or @saad-ali ?), but perhaps they can weigh in (to summarize, the question is does [the call to `MatchNodeSelectorTerms` in `CheckVolumeNodeAffinity`](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/util.go#L179) depend on ignoring node fields?)

However, sig-scheduling also needs to consider the effect on ourselves here in the Node Affinity plugin (https://github.com/kubernetes/kubernetes/pull/95531#discussion_r507879839): is it important that we are only checking for matches against `metadata.name` or do we want to consider all of the node fields?

Reply:
storage shouldn't depend on the check to be nil. I think it would be nice to have a consistent check that can be shared across all users of this field, instead of each needing to supply the list of fields to check against.

Reply:
According to [the documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#supported-fields) we should just support name and namespace (which doesn't apply to Nodes). Is there any other field supported for Nodes when using filters? Could you investigate this @damemi?
This is the opportunity to make this uniform.

Reply:
@msau42 @alculquicondor thanks for the updates, I wasn't aware of the restriction on field selectors so I think as you mentioned Aldo, only Name applies for node selectors anyway. In that case, the only usage of this field is to specifically exclude `metadata.name` from this node selector check, which seems like an oddly specific edge case. I also couldn't find any use cases of the `MatchFields` option that *didn't* use `metadata.name`.

Based on that I think we have a consensus that the extra fields are unnecessary, and I'll update this to simplify the function signature. Thanks for the feedback!